src/main/java/org/spongepowered/api/event/entity/ai/SetAttackTargetEvent.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.event.entity.ai;
+
+import org.spongepowered.api.entity.Entity;
+import org.spongepowered.api.entity.living.Agent;
+import org.spongepowered.api.event.Cancellable;
+import org.spongepowered.api.event.entity.living.TargetAgentEvent;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+/**
+ * Fires before the attack target of an {@link Agent} is set.
+ */
+public interface SetAttackTargetEvent extends TargetAgentEvent, Cancellable {
+
+    /**
+     * Gets the {@link Agent}s new target {@link Entity}.
+     *
+     * @return The target
+     */
+    Optional<Entity> getTarget();
+
+    /**
+     * Sets the {@link Agent}s new target {@link Entity}.
+     *
+     * <p>This target is set after the event when it is not cancelled.</p>
+     *
+     * @param target The target
+     */
+    void setTarget(@Nullable Entity target);
```

Comment:
Though it might be obvious to us, perhaps state what happens in the JD if null is passed (i.e. no target)?